tests/cases/unittests/commandLineParsing.ts

```diff
@@ -0,0 +1,160 @@
+            const parsed = ts.parseCommandLine(commandLine);
+            const parsedCompilerOptions = JSON.stringify(parsed.options);
+            const expectedCompilerOptions = JSON.stringify(expectedParsedCommandLine.options);
+            assert.equal(parsedCompilerOptions, expectedCompilerOptions);
+
+            const parsedErrors = parsed.errors;
+            const expectedErrors = expectedParsedCommandLine.errors;
+            assert.isTrue(parsedErrors.length === expectedErrors.length, `Expected error: ${JSON.stringify(expectedErrors)}. Actual error: ${JSON.stringify(parsedErrors)}.`);
+            for (let i = 0; i < parsedErrors.length; ++i) {
+                const parsedError = parsedErrors[i];
+                const expectedError = expectedErrors[i]; 
+                assert.equal(parsedError.code, expectedError.code, `Expected error-code: ${JSON.stringify(expectedError.code)}. Actual error-code: ${JSON.stringify(parsedError.code)}.`);
+                assert.equal(parsedError.category, expectedError.category, `Expected error-category: ${JSON.stringify(expectedError.category)}. Actual error-category: ${JSON.stringify(parsedError.category)}.`);
+            }
+
+            const parsedFileNames = parsed.fileNames;
+            const expectedFileNames = expectedParsedCommandLine.fileNames;
+            assert.isTrue(parsedFileNames.length === expectedFileNames.length, `Expected fileNames: [${JSON.stringify(expectedFileNames)}]. Actual fileNames: [${JSON.stringify(parsedFileNames)}].`);
+            for (let i = 0; i < parsedFileNames.length; ++i) {
+                const parsedFileName = parsedFileNames[i];
+                const expectedFileName = expectedFileNames[i]; 
+                assert.equal(parsedFileName, expectedFileName, `Expected filename: ${JSON.stringify(expectedFileName)}. Actual fileName: ${JSON.stringify(parsedFileName)}.`);
+            }
+        }
+
+        it("Parse single option of library flag ", () => {
+            // --lib es6 0.ts
+            assertParseResult(["--lib", "es6", "0.ts"],
+                {
+                    errors: [],
+                    fileNames: ["0.ts"],
+                    options: {
+                        lib: ["lib.es6.d.ts"]
+                    }
+                });
+        });
+
+        it("Parse multiple options of library flags ", () => {
+            // --lib es5,es6.symbol.wellknown 0.ts
+            assertParseResult(["--lib", "es5,es6.symbol.wellknown", "0.ts"],
+                {
+                    errors: [],
+                    fileNames: ["0.ts"],
+                    options: {
+                        lib: ["lib.es5.d.ts", "lib.es6.symbol.wellknown.d.ts"]
+                    }
+                });
+        });
+
+        it("Parse unavailable options of library flags ", () => {
+            // --lib es5,es7 0.ts
+            assertParseResult(["--lib", "es5,es8", "0.ts"],

```

Comment:
Ideally this is not susceptible to breaking because ES2017 will just be called ES2017. :smile: 

As a heads up, ES4 was abandoned and is available for this.